onnxruntime/python/tools/quantization/quantize.py

```diff
@@ -0,0 +1,899 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import os
+import onnx
+import onnx.numpy_helper
+import struct
+
+import numpy as np
+from onnx import onnx_pb as onnx_proto
+
+__producer__ = "onnx.quantize"
+__version__ = "0.1.0"
+onnx_domain = "ai.onnx"
+onnx_version = 10
+
+type_to_name = {
+    1: "FLOAT",
+    2: "UINT8",
+    3: "INT8",
+    4: "UINT16",
+    5: "INT16",
+    6: "INT32",
+    7: "INT64",
+    8: "STRING",
+    9: "BOOL",
+    10: "FLOAT16",
+    11: "DOUBLE",
+    12: "UINT32",
+    13: "UINT64",
+    14: "COMPLEX64",
+    15: "COMPLEX128",
+}
+
+# Quantization mode
+# IntegerOps_Static: Use IntegerOps in quantized model. Only ConvInteger and MatMulInteger ops are supported now.
+#               Use static quantization for weights and inputs/activations.
+# IntegerOps_Dynamic: Use IntegerOps in quantized model. Only ConvInteger and MatMulInteger ops are supported now.
+#               Use static quantization for weights and dynamic quantization for inputs/activations.
+# QLinearOps_Static: Use QLinearOps in quantized model. Only QLinearConv and QLinearMatMul ops are supported now.
+#               Use static quantization for weights and inputs/activations.
+# QLinearOps_Dynamic: Use QLinearOps in quantized model. Only QLinearConv and QLinearMatMul ops are supported now.
+#               Use static quantization for weights and dynamic quantization for inputs/activations.
+class QuantizationMode():
+    IntegerOps_Static = 0
+    IntegerOps_Dynamic = 1
+    QLinearOps_Static = 2
+    QLinearOps_Dynamic = 3
+
+    @staticmethod
+    def is_integer_ops_mode(mode):
+        return mode in [QuantizationMode.IntegerOps_Static, QuantizationMode.IntegerOps_Dynamic]
+
+    @staticmethod
+    def is_qlinear_ops_mode(mode):
+        return mode in [QuantizationMode.QLinearOps_Static, QuantizationMode.QLinearOps_Dynamic]
+
+    @staticmethod
+    def is_static_mode(mode):
+        return mode in [QuantizationMode.IntegerOps_Static, QuantizationMode.QLinearOps_Static]
+
+# Data Quantization mode
+# Linear_NonScaled: Quantize data using linear, non scaled tranformation.
```

Comment:
not understand, what do you mean by "non scaled" please?

Reply:
Linear_Scaled is used to quantize data into uint8 and Linear_NonScaled is used to quantize data into int8. The function quantize_data() has more explanation on this.

Reply:
Basically in scaled mode zero point is 0 and in non scaled mode it is not necessarily 0.